token/js/test/e2e-2022/tokenMetadata.test.ts

```diff
@@ -0,0 +1,444 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import type { Connection, Signer } from '@solana/web3.js';
+import { sendAndConfirmTransaction, PublicKey, Keypair, SystemProgram, Transaction } from '@solana/web3.js';
+
+import {
+    ExtensionType,
+    createInitializeMetadataPointerInstruction,
+    createInitializeMintInstruction,
+    getEmittedTokenMetadata,
+    getMintLen,
+    getTokenMetadata,
+    tokenMetadataEmit,
+    tokenMetadataInitialize,
+    tokenMetadataInitializeWithRentTransfer,
+    tokenMetadataRemoveKey,
+    tokenMetadataUpdateAuthority,
+    tokenMetadataUpdateField,
+} from '../../src';
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+
+chai.use(chaiAsPromised);
+
+const TEST_TOKEN_DECIMALS = 2;
+
+describe('Token Metadata initialization', async () => {
+    const EXTENSIONS = [ExtensionType.MetadataPointer];
+    const mintLen = getMintLen(EXTENSIONS);
+
+    let connection: Connection;
+    let payer: Signer;
+    let mint: Keypair;
+    const authority = Keypair.fromSecretKey(
+        new Uint8Array([
+            118, 177, 37, 231, 15, 88, 210, 92, 79, 231, 202, 22, 11, 15, 121, 54, 95, 229, 149, 119, 48, 177, 187, 198,
+            223, 51, 225, 74, 12, 54, 172, 36, 207, 107, 122, 208, 209, 168, 61, 177, 190, 137, 23, 156, 84, 32, 34, 82,
+            158, 176, 55, 51, 236, 66, 130, 167, 118, 31, 120, 107, 100, 192, 147, 10,
+        ])
+    );
+
+    before(async () => {
+        connection = await getConnection();
+        payer = await newAccountWithLamports(connection, 1000000000);
+    });
+
+    beforeEach(async () => {
+        mint = Keypair.generate();
+
+        const lamports = await connection.getMinimumBalanceForRentExemption(mintLen);
+
+        const transaction = new Transaction().add(
+            SystemProgram.createAccount({
+                fromPubkey: payer.publicKey,
+                newAccountPubkey: mint.publicKey,
+                space: mintLen,
+                lamports: lamports,
+                programId: TEST_PROGRAM_ID,
+            }),
+            createInitializeMetadataPointerInstruction(
+                mint.publicKey,
+                authority.publicKey,
+                mint.publicKey,
+                TEST_PROGRAM_ID
+            ),
+            createInitializeMintInstruction(
+                mint.publicKey,
+                TEST_TOKEN_DECIMALS,
+                authority.publicKey,
+                null,
+                TEST_PROGRAM_ID
+            )
+        );
+
+        await sendAndConfirmTransaction(connection, transaction, [payer, mint], undefined);
+    });
+
+    it('can successfully initialize', async () => {
+        // await expect(
```

Comment:
Whats the intended behaviour of there is insufficient funds for rent? It current gets a timeout error waiting for the block hash. Is it something we should test for?

Reply:
I'm not sure why you're experiencing a timeout. The proper error if a mint does not have enough lamports for rent should be the specific error that says "insufficient lamports for rent".

Reply:
Are we sure its suppose to throw an error? I can't figure out whats wrong.

I get the `insufficient lamports for rent` when I am creating the mint, but not when i'm just trying to initialise the metadata data in this test. The mint is created, but doesn't have enough lamports to hold the token metadata. 

From my basic understanding of the rust code - it actually doesn't throw an error, but assumes there is enough space?
https://github.com/solana-labs/solana-program-library/blob/master/token/program-2022/src/extension/token_metadata/processor.rs#L101-L101